### PR TITLE
Translation system config reading fix

### PR DESCRIPTION
**Pull request by MahBoiDeveloper**
During the porting of the Twisted Insurrection russifier to the new client version, I was surprised that the key reading algorithm for copying the game files was hardcoded to the key number after `GameFile`. Thanks to Kerbiter, I was able to quickly find the place where those keys were read and processed.  Now to make this feature work, you don't need to a specific number in keys. You can use any set of letters in the keys, as long as the keys starts with `GameFile`.

Example `[Translations]` section keys in `ClientDefinitions.ini` with **old** reading algorithm:
```ini
[Translations]
; Localized in-game fonts
GameFile0=MIX/ecache98.mix,MIX/ecache98.mix

; In-game menu text
GameFile1=Resources/language_800x600.dll,Resources/language_800x600.dll
GameFile2=Resources/language_1024x720.dll,Resources/language_1024x720.dll

; In-game mission text and units naming
GameFile3=INI/LangFS.ini,INI/LangFS.ini
GameFile4=INI/Tutorial.ini,INI/Tutorial.ini

; Missions with localized briefings
GameFile5=Missions/g0.map,Maps/Missions/g0.map
GameFile6=Missions/g1.map,Maps/Missions/g1.map
GameFile7=Missions/g10.map,Maps/Missions/g10.map
GameFile8=Missions/g11.map,Maps/Missions/g11.map
GameFile9=Missions/g12.map,Maps/Missions/g12.map
GameFile10=Missions/g13.map,Maps/Missions/g13.map
```

Example `[Translations]` section keys in `ClientDefinitions.ini` with **new** reading algorithm:
```ini
; Localized in-game fonts
GameFileFonts=MIX/ecache98.mix,MIX/ecache98.mix

; In-game menu text
GameFile01=Resources/language_800x600.dll,Resources/language_800x600.dll
GameFile0000002=Resources/language_1024x720.dll,Resources/language_1024x720.dll

; In-game mission text and units naming
GameFileUnitNames=INI/LangFS.ini,INI/LangFS.ini
GameFileMissionText=INI/Tutorial.ini,INI/Tutorial.ini

; Missions with localized briefings
GameFileMissionG00=Missions/g0.map,Maps/Missions/g0.map
GameFileMissionG01=Missions/g1.map,Maps/Missions/g1.map
GameFileMissionG10=Missions/g10.map,Maps/Missions/g10.map
GameFileMissionG11=Missions/g11.map,Maps/Missions/g11.map
GameFileMissionG12=Missions/g12.map,Maps/Missions/g12.map
GameFileMissionG13=Missions/g13.map,Maps/Missions/g13.map
```

### CODE REVIEW

**Review by Metadorius — APPROVED** [review]
Looks good, could you update to the newest develop (we switched back to net48 as the main target) and verify everything works under net48?

**Comment by MahBoiDeveloper** [comment]
PR has been update to the newest develop changes and test. My custom TI client works fine. Documentation has been updated too.

File: ClientCore/ClientConfiguration.cs [hunk 1]
```diff
@@ -278,23 +278,26 @@ private List<TranslationGameFile> ParseTranslationGameFiles()
         {
             List<TranslationGameFile> gameFiles = new();
 
-            for (int i = 0; clientDefinitionsIni.KeyExists(TRANSLATIONS, $"GameFile{i}"); i++)
+            foreach (string key in clientDefinitionsIni.GetSectionKeys(TRANSLATIONS))
             {
-                // the syntax is GameFileX=path/to/source.file,path/to/destination.file[,checked]
-                string value = clientDefinitionsIni.GetStringValue(TRANSLATIONS, $"GameFile{i}", string.Empty);
-                string[] parts = value.Split(',', StringSplitOptions.RemoveEmptyEntries).Select(s => s.Trim()).ToArray();
+                // syntax must bet GameFileX=..., where X - any text.
```
Comment (Metadorius) [inline ClientCore/ClientConfiguration.cs:283]:
```suggestion
                // the syntax is GameFileX=path/to/source.file,path/to/destination.file[,checked]
                // where X can be any text or number
```
Please don't erase info without real need.

File: ClientCore/ClientConfiguration.cs [hunk 2]
```diff
@@ -278,23 +278,26 @@ private List<TranslationGameFile> ParseTranslationGameFiles()
...
+                if (!key.StartsWith("GameFile"))
+                    continue;
+
+                string   value = clientDefinitionsIni.GetStringValue(TRANSLATIONS, key, string.Empty);
+                string[] parts = value.Split(',');
```
Comment (Metadorius) [inline ClientCore/ClientConfiguration.cs:288]:
```suggestion
                string value = clientDefinitionsIni.GetStringValue(TRANSLATIONS, key, string.Empty);
                string[] parts = value.Split(',');
```
I see the idea in the style change but I think there's more harm in this being inconsistent with the rest of client codebase than in the subjective appeal of lining up the variable declarations in this specific place. If needed such changes must be made after discussion with client maintainers/contributors and across the client globally.

Comment (MahBoiDeveloper) [reply ClientCore/ClientConfiguration.cs]: Ok, i'll keep that in mind.

**Pull request merged**